tests/cpu_bugs/xen_pv.pm

```diff
@@ -0,0 +1,418 @@
+          . " --serial pty", timeout => 3600);
+    script_run("virsh domxml-to-native xen-xl --domain ${vm_domain_name} > ${vm_domain_name}.cfg");
+    script_run("xl create \"${vm_domain_name}.cfg\"");
+    script_run('cd mitigation-testsuite');
+    $vm_ip_addr = script_output(
+        "export password=nots3cr3t; ./get-ip-from-alive-guest.expect"
+          . " \"${vm_domain_name}\"|"
+          . " grep -E -o "
+          . "\"=[[:digit:]]*\\.[[:digit:]]*\\.[[:digit:]]*\\.[[:digit:]]*=\" | tr -d \"=\"");
+    #xen_pv_guest test parameter be stored into arg
+    #$value include the check rules of current $parameter.
+    ssh_vm_cmd("cat /proc/cmdline | grep \"mitigations=auto\"");
+    ssh_vm_cmd("sed -i '/GRUB_CMDLINE_LINUX_DEFAULT=/s/mitigations=[a-z,]*/\\ /' /etc/default/grub");
+    config_and_reboot();
+    while (my ($arg, $dict) = each %$hash) {
+        while (my ($key, $value) = each %$dict) {
+            my $parameter = $arg . '=' . $key;
+            ssh_vm_cmd("sed -i -e 's/GRUB_CMDLINE_LINUX=\"\"/GRUB_CMDLINE_LINUX=\"$parameter\"/' /etc/default/grub");
+            config_and_reboot();
+            my $ret = do_check($value);
+            if ($ret ne 0) {
+                record_info('ERROR', "$parameter test is failed.", result => 'fail');
+            }
+            record_info('INFO', "$parameter test is finished.");
+            ssh_vm_cmd("sed -i -e 's/GRUB_CMDLINE_LINUX=\"$parameter\"/GRUB_CMDLINE_LINUX=\"\"/' /etc/default/grub");
+            config_and_reboot();
+        }
+    }
+
+
+}
+
+sub config_and_reboot {
+    #my $timeout = @_;
+    ssh_vm_cmd("grub2-mkconfig -o /boot/grub2/grub.cfg");
+    ssh_vm_cmd("poweroff");
+    script_run('xl list');
+    script_run('xl list');
+    script_run("xl create \"${vm_domain_name}.cfg\"");
+    script_run('echo Now I am waiting for the vm to reboot');
+    script_run("sleep 60");
+    ssh_vm_cmd("cat /proc/cmdline");
+}
+
+sub ssh_vm_cmd {
+    my $cmd = shift;
+    my $ret = script_run("sshpass -p ${qa_password} ssh -qy root\@${vm_ip_addr} $cmd");
+    return $ret;
+}
+
+sub run {
+    my $self = @_;
+    select_console 'root-console';
+    die "platform mistake, This system is not running as Dom0." if script_run("test -d /proc/xen");
+    do_test($self, $mitigations_list);
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+    select_console 'root-console';
+    script_run(
+"md /tmp/upload_mitigations; cp " . $Mitigation::syspath . "* /tmp/upload_mitigations; cp /proc/cmdline /tmp/upload_mitigations; lscpu >/tmp/upload_mitigations/cpuinfo; tar -jcvf /tmp/upload_mitigations.tar.bz2 /tmp/upload_mitigations"
+    );
+    script_run("xl info");
+    upload_logs '/tmp/upload_mitigations.tar.bz2';
+    script_run("rm -rf /tmp/upload_mitigations");
+    script_run("rm -rf /tmp/upload_mitigations.tar.bz2");
+    remove_grub_cmdline_settings('xpti=[a-z,]*');
+    remove_grub_cmdline_settings('spec-ctrl=[a-z,-]*');
+    remove_grub_cmdline_settings('pv-l1tf=[a-z,]*');
```

Comment:
These operation is happen at host, not guest. so If you need clean up Guests, you have to change to ssh_vm_cmd() like functions.